portafly/src/pages/Login.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react'
+import { Redirect } from 'react-router-dom'
+import { useAuth } from 'auth'
+import { useTranslation } from 'i18n/useTranslation'
+import { useDocumentTitle } from 'components/util'
+import {
+  ActionGroup,
+  Button,
+  Form,
+  FormGroup,
+  TextInput,
+  LoginPage
+} from '@patternfly/react-core'
+import brandImg from 'assets/logo.svg'
+
+interface ILoginForm {
+  token: string
+  onChange: (v: string) => void
+  onClick: () => void
+  t: (v: string) => string
+}
+
+const LoginForm = (
+  {
+    token,
+    onChange,
+    onClick,
+    t
+  }: ILoginForm
+) => (
+  <Form>
+    <FormGroup
+      label={t('form_label_token')}
+      isRequired
+      fieldId="simple-form-name"
+    >
+      <TextInput
+        isRequired
+        type="text"
+        id="token"
+        name="token"
+        value={token}
+        onChange={onChange}
+      />
+      <ActionGroup>
+        <Button onClick={onClick} variant="primary">{t('login_submit')}</Button>
+        <Button variant="secondary">{t('login_cancel')}</Button>
+      </ActionGroup>
+    </FormGroup>
+  </Form>
+)
+
+
+const Login = () => {
+  const [isLoggedIn, setLoggedIn] = useState(false)
+  const [token, setToken] = useState('')
+  const { setAuthToken } = useAuth()
+  const { t } = useTranslation('login')
+  useDocumentTitle(t('page_title'))
+
+  function postLogin() {
+    setAuthToken(token)
+    setLoggedIn(true)
+  }
+
+  if (isLoggedIn) {
+    return <Redirect to="/" />
+  }
+
+  return (
+    <LoginPage
+      brandImgSrc={brandImg}
+      brandImgAlt={t('brand_img_alt')}
+      loginTitle={t('login_title')}
+      loginSubtitle={t('login_subtitle')}
+    >
+      <LoginForm
```

Comment:
I would rather define this as a const inside `Login` and not passing props (specially t looks weird)

Reply:
It was like that before, was trying to comply with the 25 lines per function rule, still failed.

Reply:
I agree that passing `t` is weird... I could instantiate the hook again inside that component. WDYT? any other solution would be weirder (or go back to a 44+ lines function)

Reply:
> It was like that before, was trying to comply with the 25 lines per function rule, still failed.

Am I missing something or since when is this an issue? I mean, it's defo not a problem to extract consts and functions in the same file, as long as they make sense and they cannot be extracted. But for this particular case that the _inner_ component relies on internal variables/functions I think it's better so simply put it inside. I believe it's going to be literally impossible to maintain a 25 lines limit per function.

Reply:
> I agree that passing `t` is weird... I could instantiate the hook again inside that component. WDYT? any other solution would be weirder (or go back to a 44+ lines function)

I think instantiating the hook inside each component is the way to go, although I still don't know if that's going to be a problem performance-wise. It's what I'm doing for my PR.